framework/src/vectorpostprocessors/VectorPostprocessorData.C

```diff
@@ -70,7 +70,8 @@ VectorPostprocessorData::getVectorPostprocessorHelper(const VectorPostprocessorN
                                                       bool contains_complete_history)
 {
   // Retrieve or create the data structure for this VPP
-  auto vec_it_pair = _vpp_data.emplace(vpp_name, VectorPostprocessorVectors());
+  auto vec_it_pair = _vpp_data.emplace(
+      std::piecewise_construct, std::forward_as_tuple(vpp_name), std::forward_as_tuple());
```

Comment:
BTW - std::move() causes the compiler to complain here saying it prevents copy elision. Implementing the move constructors doesn't necessarily fix the problem either. This admittedly ugly solution is the only one that will actually create the pair in-place. I tacked this on as an extra commit, but am willing to remove if it's deemed too ugly.